### PR TITLE
Filesystem

### PR DESCRIPTION
**Pull request by krulis-martin**

### CODE REVIEW

File: app/V1Module/presenters/AssignmentsPresenter.php [hunk 1]
```diff
@@ -698,7 +704,7 @@ public function actionDownloadBestSolutionsArchive(string $id)
 
             $studentDir = Strings::webalize("{$student->getLastName()}_{$student->getFirstName()}_{$student->getId()}");
             foreach ($solution->getSolution()->getFiles() as $file) {
-                $files[$file->getLocalFilePath()] = $studentDir . "/" . $file->getName();
+                $files[$studentDir . "/" . $file->getName()] = $file->getFile($this->fileStorage);
```
Comment (Neloop) [inline app/V1Module/presenters/AssignmentsPresenter.php:707]: > $file->getFile($this->fileStorage

this seems like a bit reversed logic

Comment (krulis-martin) [reply app/V1Module/presenters/AssignmentsPresenter.php:707]: Depends on point of view, I suppose. However, the file returned from storage is wrapped in ImmutableFile object (abstraction over files), so we have no choice. Furthermore, I see logic in having the presentation name as key and the actual file (object in this case) as value.

File: app/http/ZipFilesResponse.php [hunk 2]
```diff
@@ -54,9 +49,18 @@ private function compress()
...
-            if ($zip->addFile($localPath, $name) !== true) {
-                throw new ApiException("Error while adding file to archive");
+        foreach ($this->files as $name => $file) {
+            if ($file instanceof IImmutableFile) {
+                try {
+                    $file->addToZip($zip, $name);
+                } catch (FileStorageException $e) {
+                    throw new ApiException("Error while adding file to archive");
+                }
+            } else {
```
Comment (Neloop) [inline app/http/ZipFilesResponse.php:59]: Is this else really needed, is there situation where this would be used?

Comment (krulis-martin) [reply app/http/ZipFilesResponse.php:59]: You are right, it is not used anywhere. But I was thinking a bit ahead trying to make it reusable...

**Review by Neloop — APPROVED** [review]

**Pull request merged**